XCTestHTMLReport/XCTestHTMLReport
# Support for Mixed status tests and repetition policies

Background:
Xcode 13 introduced retry policies which allowed a single test case to be reported with a "Mixed" status depending on the policy. Currently, HTML reports count each iteration as a separate test and will report an incorrect number of test cases and statuses.

After merging davidahouse/XCResultKit#40 we now have access to `RepetitionPolicySummary` and can now nest multiple iterations under one test.

Changes:
- Add `Mixed` to all statuses and related logic, including page filters
- Make the `Test` object a protocol, conforms by new `TestGroup`, `TestCase`, and `Iteration` objects
- Update Run, Summary, HTML, and JUnit to consume the above objects
- **Modify HTML templates to automatically nest >1 iterations under its respective test case**
  -  **For test cases with 1 iteration or no repetition policy, keep activities directly under the test case**
- Update unit tests, adding a separate test run with a repetition policy
- General auto-formatting with nicklockwood/SwiftFormat


Testing:
- Verify no changes to status reporting or UI layout for tests without repetition policies
- Verify a test with a repetition policy but only one iteration will be displayed like the above
- Verify a test with a repetition policy and multiple iterations lists all iterations under the respective test case
- Verify a test with a repetition policy and multiple iterations containing at least one failure and at least one success will display alongside each iteration, and the respective test case show a "Mixed" status
- Verify test counts accurately reflect the number of test cases and not iterations
